plugins/workload-management/src/test/java/org/opensearch/plugin/wlm/QueryGroupTestUtils.java

```diff
@@ -0,0 +1,125 @@
+    public static final String NAME_NONE_EXISTED = "sandbox_none_existed";
+    public static final String MONITOR = "monitor";
+    public static final long TIMESTAMP_ONE = 4513232413L;
+    public static final long TIMESTAMP_TWO = 4513232415L;
+    public static final QueryGroup queryGroupOne = builder().name(NAME_ONE)
+        ._id(_ID_ONE)
+        .mode(MONITOR)
+        .resourceLimits(Map.of("jvm", 0.3))
+        .updatedAt(TIMESTAMP_ONE)
+        .build();
+
+    public static final QueryGroup queryGroupTwo = builder().name(NAME_TWO)
+        ._id(_ID_TWO)
+        .mode(MONITOR)
+        .resourceLimits(Map.of("jvm", 0.6))
+        .updatedAt(TIMESTAMP_TWO)
+        .build();
+
+    public static final Map<String, QueryGroup> queryGroupMap = Map.of(NAME_ONE, queryGroupOne, NAME_TWO, queryGroupTwo);
+
+    public static List<QueryGroup> queryGroupList() {
+        return List.of(queryGroupOne, queryGroupTwo);
+    }
+
+    public static ClusterState clusterState() {
+        final Metadata metadata = Metadata.builder().queryGroups(new HashSet<>(queryGroupList())).build();
+        return ClusterState.builder(new ClusterName("_name")).metadata(metadata).build();
+    }
+
+    public static Settings settings() {
+        return Settings.builder().build();
+    }
+
+    public static ClusterSettings clusterSettings() {
+        return new ClusterSettings(settings(), ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+    }
+
+    public static QueryGroupPersistenceService queryGroupPersistenceService() {
+        ClusterService clusterService = new ClusterService(settings(), clusterSettings(), mock(ThreadPool.class));
+        return new QueryGroupPersistenceService(clusterService, settings(), clusterSettings());
+    }
+
+    public static List<Object> prepareSandboxPersistenceService(List<QueryGroup> queryGroups) {
+        Metadata metadata = Metadata.builder().queryGroups(new HashSet<>(queryGroups)).build();
+        Settings settings = Settings.builder().build();
+        ClusterSettings clusterSettings = new ClusterSettings(settings, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        ClusterService clusterService = new ClusterService(settings, clusterSettings, mock(ThreadPool.class));
+        ClusterState clusterState = ClusterState.builder(new ClusterName("_name")).metadata(metadata).build();
+        QueryGroupPersistenceService queryGroupPersistenceService = new QueryGroupPersistenceService(
+            clusterService,
+            settings,
+            clusterSettings
+        );
+        return List.of(queryGroupPersistenceService, clusterState);
+    }
+
+    public static void compareResourceLimits(Map<String, Object> resourceLimitMapOne, Map<String, Object> resourceLimitMapTwo) {
+        for (String resourceName : resourceLimitMapOne.keySet()) {
+            assertTrue(resourceLimitMapTwo.containsKey(resourceName));
+            assertEquals(resourceLimitMapOne.get(resourceName), resourceLimitMapTwo.get(resourceName));
```

Comment:
We can replace this with

```suggestion
        assertTrue(resourceLimitMapOne.keySet().containsAll(resourceLimitMapTwo.keySet()));
        assertTrue(resourceLimitMapOne.values().containsAll(resourceLimitMapTwo.values()));
```